frontend/src/App.tsx

```diff
@@ -58,6 +84,280 @@ interface IRouteGroup {
     routes: IRoute[]
 }
 
+function api<T>(url: string, headers?: Record<string, unknown>): Promise<T> {
+    return fetch(url, headers).then((response) => {
+        if (!response.ok) {
+            throw new Error(response.statusText)
+        }
+        return response.json() as Promise<T>
+    })
+}
+
+function apiNoJSON(url: string, headers?: Record<string, unknown>): Promise<unknown> {
+    return fetch(url, headers).then((response) => {
+        if (!response.ok) {
+            throw new Error(response.statusText)
+        }
+        return response.text() as Promise<unknown>
+    })
+}
+
+function launchToOCP(urlSuffix: string) {
+    api<{ data: { consoleURL: string } }>(
+        '/multicloud/api/v1/namespaces/openshift-config-managed/configmaps/console-public/'
+    )
+        .then(({ data }) => {
+            window.open(`${data.consoleURL}/${urlSuffix}`)
+        })
+        .catch((error) => {
+            // eslint-disable-next-line no-console
+            console.error(error)
+        })
+}
+
+function checkOCPVersion(switcherExists: (arg0: boolean) => void) {
+    if (process.env.NODE_ENV === 'test') return
+    api<{ gitVersion: string }>('/multicloud/version/')
+        .then(({ gitVersion }) => {
+            if (parseFloat(gitVersion.substr(1, 4)) >= 1.2) {
+                switcherExists(true)
+            } else {
+                switcherExists(false)
+            }
+        })
+        .catch((error) => {
+            // eslint-disable-next-line no-console
+            console.error(error)
+            switcherExists(false)
+        })
+}
+
+function UserDropdownToggle() {
+    const [name, setName] = useState<string>('loading...')
+
+    useEffect(() => {
+        // Get the username from the console backend
+        const resp = getUsername()
+        resp.promise
+            .then((payload) => {
+                payload && payload.body && payload.body.username ? setName(payload.body.username) : setName('undefined')
+            })
+            .catch((error) => {
+                // eslint-disable-next-line no-console
+                console.error(error)
```

Comment:
Debugging code?

Reply:
No, in case the getUsername response had an error rather than an actual response (should be a pretty rare case if it can even occur) this catch is just to be sure the error ends up in the console, but making sure the name value is set to some string.